artiaa_auth/src/error.rs

```diff
@@ -0,0 +1,58 @@
+use std::{fmt, io, path::PathBuf};
+pub type ArtifactoryAuthResult<T> = Result<T, ArtifactoryAuthError>;
+
+#[derive(Debug)]
+pub enum ArtifactoryAuthError {
+    FileParse { source: String, path: PathBuf },
+    Read { source: io::Error, path: PathBuf },
+    Write { source: io::Error, path: PathBuf },
+}
+
+impl ArtifactoryAuthError {
+    pub fn auth_parsing<P: Into<PathBuf>, S: Into<String>>(auth_path: P, source: S) -> Self {
+        Self::FileParse {
+            source: source.into(),
+            path: auth_path.into(),
+        }
+    }
+    pub fn read_error<P: Into<PathBuf>>(source: io::Error, path: P) -> Self {
+        Self::Read {
+            source,
+            path: path.into(),
+        }
+    }
+    pub fn write_error<P: Into<PathBuf>>(source: io::Error, path: P) -> Self {
+        Self::Write {
+            source,
+            path: path.into(),
+        }
+    }
+}
+
+const ARTIFACTORY_AUTH_HELP: &str = include_str!("../../resources/artiaa-format.json");
+
+impl fmt::Display for ArtifactoryAuthError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::FileParse { source, path } => write!(
+                f,
+                "unable to parse Artifactory authentication file (at {}): {}\n\nAn Artifactory authentication file looks like this\n\n{}",
+                path.display(),
+                source,
+                ARTIFACTORY_AUTH_HELP
+            ),
```

Comment:
Is this the right format we want to show users as an example

Reply:
Not exactly, since that's a json schema definition. For now, though, you could just change the wording to something like "An Artifactory authentication file should match this schema:"